File: server/TasksManager.API/Models/Enums.cs
namespace TasksManager.API.Models;

public enum GoalType
{
    Finite = 0,
    Ongoing = 1
}

public enum RecurrenceType
{
    None    = 0,
    Daily   = 1,
    Weekly  = 2,
    Monthly = 3
}

public enum Priority
{
    Low = 0,
    Medium = 1,
    High = 2,
    Critical = 3
}

public enum Difficulty
{
    Easy = 0,
    Medium = 1,
    Hard = 2
}

public enum ExecutionType
{
    Quick  = 0,
    Short  = 1,
    Medium = 2,
    Long   = 3
}

public enum TaskNature
{
    Action      = 0,  // something you DO yourself
    Meeting     = 1,  // scheduled meeting with others
    Appointment = 2,  // fixed external appointment (doctor, dentist, etc.)
}

public enum ItemStatus
{
    Open        = 0,  // default – not yet completed
    Completed   = 1,  // user marked as done
    Archived    = 2,  // manually archived
    Missed      = 3,  // time-sensitive task that passed without completion
    CarriedOver = 4,  // past-due non-time-sensitive task brought forward
}

public enum ListType
{
    Checklist   = 0,  // default – generic checklist
    Shopping    = 1,  // smart shopping list with departments + suggestions
    Notes       = 2,  // free-form notes
    Ideas       = 3,  // idea collection
    Equipment   = 4,  // packing / equipment list
    CookingPlan = 5,  // meal / cooking planning list
}

public enum ShoppingDepartment
{
    FruitsAndVegetables = 0,
    Dairy               = 1,
    MeatAndFish         = 2,
    Bakery              = 3,
    Pantry              = 4,
    Frozen              = 5,
    Cleaning            = 6,
    Disposable          = 7,
    Baby                = 8,
    Other               = 9,
}

public enum ShoppingItemType
{
    Regular    = 0,  // always suggest again after being bought
    Occasional = 1,  // suggest again after X days
    OneTime    = 2,  // never auto-suggest
}

/// <summary>Which meal slot a cooking item belongs to.</summary>
public enum MealSlot
{
    None            = 0,
    // Shabbat meals
    FridayDinner    = 1,   // ערב שבת
    SaturdayMorning = 2,   // בוקר שבת
    Additions       = 3,   // תוספות (חלות, מיצים...)
    ThirdMeal       = 4,   // סעודה שלישית
    // Weekday categories
    MainDish        = 10,  // עיקרי
    Side            = 11,  // תוספת
    Salad           = 12,  // סלט
    Dessert         = 13,  // קינוח
    Other           = 99,
}

/// <summary>How a cooking list is structured.</summary>
public enum CookingMode
{
    Regular = 0,  // weekday cooking – optional category grouping
    Shabbat = 1,  // Shabbat cooking – divided by meal slots
}

File: client/src/types/enums.ts
// ─────────────────────────────────────────────────────────────────────────────
// Enums
// All string values map to i18n translation keys where appropriate.
// ─────────────────────────────────────────────────────────────────────────────

/** Whether a goal has a definite end-state or is maintained indefinitely */
export enum GoalType {
  /** Has a clear completion point (e.g. "Learn Python basics") */
  Finite  = 'finite',
  /** Ongoing habit / maintenance (e.g. "Stay healthy") */
  Ongoing = 'ongoing',
}

/** Broad category used for goal icon / colour coding */
export enum GoalCategory {
  Home     = 'home',
  Work     = 'work',
  Health   = 'health',
  Business = 'business',
  Python   = 'python',
  Hobby    = 'hobby',
  Personal = 'personal',
}

/**
 * How urgently a task should be addressed.
 * Maps to translation key `priority.<value>`.
 */
export enum Priority {
  Low      = 'low',
  Medium   = 'medium',
  High     = 'high',
  Critical = 'critical',
}

/**
 * Perceived difficulty of a task.
 * Maps to translation key `difficulty.<value>`.
 */
export enum Difficulty {
  Easy   = 'easy',
  Medium = 'medium',
  Hard   = 'hard',
}

/**
 * Rough estimate of how long a task takes to execute.
 * Used to suggest "quick win" tasks.
 * Maps to translation key `executionType.<value>`.
 */
export enum ExecutionType {
  /** Under 2 minutes */
  Quick  = 'quick',
  /** 2–10 minutes */
  Short  = 'short',
  /** 10–30 minutes */
  Medium = 'medium',
  /** 30+ minutes */
  Long   = 'long',
}

/**
 * How a task should repeat after completion.
 * Maps to translation key `recurrence.<value>`.
 */
export enum RecurrenceType {
  None    = 'none',
  Daily   = 'daily',
  Weekly  = 'weekly',
  Monthly = 'monthly',
}

/**
 * The nature / category of a task.
 * Maps to translation key `taskNature.<value>`.
 */
export enum TaskNature {
  /** Something you do yourself */
  Action      = 'action',
  /** Scheduled meeting with others */
  Meeting     = 'meeting',
  /** Fixed external appointment (doctor, dentist, admin, etc.) */
  Appointment = 'appointment',
}

/**
 * Lifecycle status of a task.
 * Maps to translation key `taskStatus.<value>`.
 */
export enum TaskStatus {
  Open        = 'open',
  Completed   = 'completed',
  Archived    = 'archived',
  /** Past-due time-sensitive task that was not completed */
  Missed      = 'missed',
  /** Past-due task carried forward to today */
  CarriedOver = 'carriedOver',
}

/**
 * The type / purpose of a personal list.
 * Maps to translation key `listType.<value>`.
 */
export enum ListType {
  Checklist   = 'checklist',
  Shopping    = 'shopping',
  Notes       = 'notes',
  Ideas       = 'ideas',
  Equipment   = 'equipment',
  CookingPlan = 'cookingplan',
}

/**
 * Supermarket / store department for grouping shopping items.
 * Maps to translation key `shoppingDepartment.<value>`.
 */
export enum ShoppingDepartment {
  FruitsAndVegetables = 'fruitsAndVegetables',
  Dairy               = 'dairy',
  MeatAndFish         = 'meatAndFish',
  Bakery              = 'bakery',
  Pantry              = 'pantry',
  Frozen              = 'frozen',
  Cleaning            = 'cleaning',
  Disposable          = 'disposable',
  Baby                = 'baby',
  Other               = 'other',
}

/**
 * How often a shopping item is purchased — drives smart suggestion logic.
 * Maps to translation key `shoppingItemType.<value>`.
 */
export enum ShoppingItemType {
  /** Always suggest again after being bought */
  Regular    = 'regular',
  /** Suggest again after X days (configurable per list) */
  Occasional = 'occasional',
  /** Never auto-suggest */
  OneTime    = 'oneTime',
}

/** Meal slot for a cooking item — Shabbat slots or weekday categories. */
export enum MealSlot {
  None            = 'none',
  // Shabbat meals
  FridayDinner    = 'fridaydinner',
  SaturdayMorning = 'saturdaymorning',
  Additions       = 'additions',
  ThirdMeal       = 'thirdmeal',
  // Weekday categories
  MainDish        = 'maindish',
  Side            = 'side',
  Salad           = 'salad',
  Dessert         = 'dessert',
  Other           = 'other',
}

/** Whether a cooking list is structured for Shabbat or regular weekday cooking. */
export enum CookingMode {
  Regular = 'regular',
  Shabbat = 'shabbat',
}

